addons/pos_event/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -49,6 +53,34 @@ patch(ProductScreen.prototype, {
             return;
         }
 
+        if (event.is_multi_slots) {
+            const slotId = await makeAwaitable(this.dialog, EventSlotSelectionPopup, {
+                event: event,
+            });
+            if (!slotId) {
+                return;
+            }
+            // Replace tickets by their related slot tickets
+            const ticketIds = ticketResult.map((r) => r.ticket_id.id);
+            const parentToSlotTicketMapping = {};
+            this.pos.models["event.event.ticket"]
+                .filter(
+                    (ticket) =>
+                        ticket.slot_id?.id == slotId &&
+                        ticket.parent_ticket_id &&
+                        ticketIds.includes(ticket.parent_ticket_id.id)
+                )
+                .forEach((slotTicket) => {
+                    parentToSlotTicketMapping[slotTicket.parent_ticket_id.id] = slotTicket;
+                });
+            ticketResult.forEach((res) => {
+                const ticketId = res.ticket_id.id;
+                if (ticketId in parentToSlotTicketMapping) {
+                    res.ticket_id = parentToSlotTicketMapping[ticketId];
+                }
+            });
```

Comment:
Ah yes, we could have different tickets in one shot, hence the various forEach. Good catch!

Reply:
should we return if any is not in the mapping? (should not happen though)